cli/src/cli.rs

```diff
@@ -739,8 +739,14 @@ pub fn get_blockhash_fee_calculator(
         } else {
             (blockhash, rpc_client.get_recent_blockhash()?.1)
         }
-    } else {
+    } else if !sign_only {
         rpc_client.get_recent_blockhash()?
+    } else {
+        return Err(CliError::BadParameter(format!(
```

Comment:
How about using https://docs.rs/clap/2.33.0/clap/struct.Arg.html#method.requires ?

Reply:
#7982 😉 

It unfortunately won't prevent us from writing incorrect tests, as I've fixed up here. Maybe `panic!()` would be more appropriate?

Reply:
Ha nice! I guess ideally it wouldn't be possible to write incorrect tests, something like:
```rust
enum SignMethod {
  Online,
  Offline(Hash, bool)
}
```

Reply:
Getting there!
https://github.com/solana-labs/solana/blob/c13ab9f14ec136f94e89778e456150f956c8c864/cli/src/cli.rs#L77-L82

Reply:
@t-nelson cool, I'd prefer that approach over the current approach

Reply:
Fully reworked as per request 😉

Reply:
Ha great! Looks like a step in the right direction!